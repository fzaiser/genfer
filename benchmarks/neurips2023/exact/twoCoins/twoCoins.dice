let firstCoin  = flip 0.5 in 
let secondCoin  = flip 0.5 in
let bothHeads  = (firstCoin && secondCoin) in
let tmp = observe !bothHeads in
firstCoin
