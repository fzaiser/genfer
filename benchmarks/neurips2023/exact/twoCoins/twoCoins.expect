Support is a subset of: {0, ..., 1}

Computing moments...
Total measure:             Z = 0.75
Expected value:            E = 0.3333333333333333
Standard deviation:        Ïƒ = 0.4714045207910317
Variance:                  V = 0.2222222222222222
Skewness (3rd std moment): S = 0.7071067811865475
Kurtosis (4th std moment): K = 1.4999999999999984

Computing probabilities up to 2...
Unnormalized: p(0)     = 0.5
Normalized:   p(0) / Z = 0.6666666666666666
Unnormalized: p(1)     = 0.25
Normalized:   p(1) / Z = 0.3333333333333333
Unnormalized: p(n)     <= 0.0 for all n >= 2
Normalized:   p(n) / Z <= 0.0 for all n >= 2
