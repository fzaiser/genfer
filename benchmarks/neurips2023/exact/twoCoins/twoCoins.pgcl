nat firstCoin;
nat secondCoin;
nat bothHeads;

firstCoin := bernoulli(1/2);
secondCoin := bernoulli(1/2);
if(firstCoin = 1 & secondCoin = 1) {
    bothHeads := 1;
} else {
    bothHeads := 0
}
observe(bothHeads = 0);

?Pr[firstCoin];
