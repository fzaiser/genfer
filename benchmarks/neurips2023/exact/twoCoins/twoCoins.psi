// flags: --dp
def main(){ // bothHeads
    firstCoin := flip(1/2);
    secondCoin := flip(1/2);
    bothHeads := firstCoin && secondCoin;
    observe(bothHeads == 0);
    return (firstCoin);
}
