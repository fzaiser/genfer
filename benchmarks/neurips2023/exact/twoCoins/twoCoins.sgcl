firstCoin ~ Bernoulli(1/2);
secondCoin ~ Bernoulli(1/2);
if firstCoin=1 && secondCoin = 1 {
    bothHeads := 1;
} else {
    bothHeads := 0;
}
observe bothHeads = 0;

return firstCoin;
