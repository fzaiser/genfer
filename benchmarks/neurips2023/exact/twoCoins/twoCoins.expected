0.33333333
p(1) / Z = 1/3
1/3·δ(1)[firstCoin]
1/3*firstCoin
